scripts/2009/07/thumbnail.py

```diff
@@ -3,84 +3,8 @@
 Usage:
 
     $ python thumbnail.py f1.warc f2.warc f3.warc
-"""
-import sys, os
-import Image
-from cStringIO import StringIO
-import time
-import tarfile
-import web
-
-import warc
-
-def makedirs(d):
-    if not os.path.exists(d):
-        os.makedirs(d)
-
-def make_thumbnail(record):
-    """Make small and medium thumbnails of given record."""
-    id = record.get_header().subject_uri.split('/')[-1].split('.')[0]
-    id = "%010d" % int(id)
-    path = "/".join([id[0:3], id[3:6], id[6:9]])
-
-    data = record.get_data()
-    image = Image.open(StringIO(data))
-
-    sizes = dict(S=(116, 58), M=(180, 360), L=(500, 500))
-
-    yield id + "-O.jpg", data
-
-    for size in "SML":
-        imgpath = "%s-%s.jpg" % (id, size)
-        try:
-            if image.mode != 'RGB':
-                image = image.convert('RGB')
 
-            thumbnail = StringIO()
-            image.resize(sizes[size], resample=Image.ANTIALIAS).save(thumbnail, format='jpeg')
-            yield imgpath, thumbnail.getvalue()
-        except Exception, e:
-            print 'ERROR:', id, str(e)
-            sys.stdout.flush()
-
-def add_image(tar, name, data, mtime=None):
-    tarinfo = tarfile.TarInfo(name)
-    tarinfo.mtime = mtime or int(time.time())
-    tarinfo.size = len(data)
-    file = StringIO(data)
-    tar.addfile(tarinfo, file)
-
-def read(warc_files):
-    for w in warc_files:
-        print time.asctime(), w
-        sys.stdout.flush()
-        reader = warc.WARCReader(open(w))
-        for r in reader.read():
-            yield r
-
-def process(warcfile, dir):
-    outpath = dir + '/' + os.path.basename(warcfile).replace('.warc', '.tar')
-    print 'process', warcfile, dir, outpath
-    tar = tarfile.TarFile(outpath, 'w')
-
-    def f(records):
-        for r in records:
-            timestamp = time.mktime(time.strptime(r.get_header().creation_date, '%Y%m%d%H%M%S'))
-            try:
-                for name, data in make_thumbnail(r):
-                    yield name, data, timestamp
-            except Exception, e:
-                print >> sys.stderr, str(e)
-
-    for name, data, timestamp in f(read([warcfile])):
-        add_image(tar, name, data, mtime=timestamp)
-    tar.close()
-
-def main(dir, *warc_files):
-    if not os.path.exists(dir):
-        os.makedirs(dir)
-    for w in warc_files:
-        process(w, dir)
-
-if __name__ == "__main__":
-    main(*sys.argv[1:])
+XXX: This file has been archived:
```

Comment:
I'm not crazy over this pattern; all the files inside `scripts/20*` are considered archived; they're excluded from linting/py3 validation/etc. Having a file like this implies that they are _not_ all archived and that we need to support them in some way (which we have not been :P ). I know there are a few scripts inside these folders that might still be being used, but we should find those and move them out, and leave the `scripts/20*` files to be archived/deprecated.

Reply:
That makes sense. Personally, I would prefer to see old, invalid/broken Python code removed from the repo, since it might confuse new contributors or new code linting tools. I'm OK with any position on this topic, though.

Reply:
> scripts/20* should kind of be "frozen"

I have said it numerous times, these files should be removed from this repo.  We have to make too many exceptions in linting, testing, Makefile, CI, search and that only slows us down.  These files should be moved to another repo and that repo should be set to archieved.

Reply:
👍 Strong agree. The only problem is that some of these files are being used at random in our prod environment and are hard to pin down where they're being. Actually, I would be ok deleting this file _entirely_ in this case since we are certain it's not being used.

Reply:
I re-opened the issue for this #1449

Reply:
Let's make the decision the move **unused** scripts to another location (e.g. an Archive.org item) and let's make sure this is explicit/documented in a/our README.md within scripts/

For now, I'm landing this issue to keep things moving; priorities.